cmd/rootMain.go

```diff
@@ -122,16 +117,77 @@ Try 'help'`)
 	)
 	comm.Dir = mountpath
 
+	// Inspired by https://blog.nelhage.com/2011/02/changing-ctty/. After the child shell has
+	// started, we re-parent the Wash daemon to be in a child session of the shell so that when Wash
+	// prompts for input it's within the TTY of that terminal.
+	if startErr := comm.Start(); startErr != nil {
+		cmdutil.ErrPrintf("%v\n", startErr)
+		return exitCode{1}
+	}
+
+	codeCh := make(chan int)
+	go func() {
+		if runErr := comm.Wait(); runErr != nil {
+			if exitErr, ok := runErr.(*exec.ExitError); ok {
+				codeCh <- exitErr.ExitCode()
+			} else {
+				cmdutil.ErrPrintf("%v\n", runErr)
+				codeCh <- 1
+			}
+		} else {
+			codeCh <- 0
+		}
+		close(codeCh)
+	}()
+
 	var exit exitCode
-	if runErr := comm.Run(); runErr != nil {
-		if exitErr, ok := runErr.(*exec.ExitError); ok {
-			exit.value = exitErr.ExitCode()
+	killCommand := func() exitCode {
+		if err := comm.Process.Kill(); err != nil {
+			cmdutil.ErrPrintf("Couldn't stop child shell: %v\n", err)
 		}
-		cmdutil.ErrPrintf("%v\n", runErr)
-		exit.value = 1
+		return exitCode{1}
 	}
 
-	srv.Stop()
+	washPid := os.Getpid()
+	pid := comm.Process.Pid
+	for {
```

Comment:
Because the process group of the child process is changed after it starts. So we need to wait until it's changed. See comment at line 171.

Reply:
So it sounds like the child process' process group is changed precisely when it becomes the leader of its own process group? If so, that comment doesn't really make that clear without some knowledge of process groups. Would be nice to reword it, and maybe move it up to the start of the `for` loop since that's where my Q. was.

Reply:
ok